web/ui/react-app/src/pages/flags/Flags.tsx

```diff
@@ -14,18 +22,107 @@ interface FlagsProps {
   data?: FlagMap;
 }
 
+const compareAlphaFn = (keys: boolean, reverse: boolean) => (
+  [k1, v1]: [string, string],
+  [k2, v2]: [string, string]
+): number => {
+  const a = keys ? k1 : v1;
+  const b = keys ? k2 : v2;
+  const reverser = reverse ? -1 : 1;
+  return reverser * a.localeCompare(b);
+};
+
+const getSearchMatches = (input: string, flagList: string[]) => {
+  return fuz.filter(input.replace(new RegExp(`/${flagSeparator}/g`), ''), flagList);
```

Comment:
It seems you can't combine the `/.../g` literal regex form with the string form as an input to `new RegExp()`. In that case, the substitution doesn't work for me.

I think you wanted:

```typescript
new RegExp(flagSeparator, 'g')
```

But: I think we only want to replace the first occurrence of the separator (so no `g` flag), because flag values may contain `||`, but flag names cannot. So replacing the first occurrence is guaranteed to catch the separator only.

But then you don't even need a `RegExp` anymore, you can just do:

```suggestion
  return fuz.filter(input.replace(flagSeparator, ''), flagList);
```

Reply:
And: do we even want to remove this at all for finding the search matches? Like, should a search string be able to span from a flag name *into* the flag value?

Reply:
well on my side, I think searching only the flag name would be enough and it simplify this kind of 'hack' too

Reply:
perhaps we can remove it for now, so you can only search the flag name and let see how people are reacting. WDYT @juliusv ?

Reply:
ok I missunderstood the initial point of @juliusv. Sry about that.

Reply:
should be good now